### ISSUE TITLE
@stylistic/declaration-colon-space-before fix produces unparsable code after an inline comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#88]
Found while fixing #63, by sweeping every rule that deletes a trailing whitespace run from a `raws` string. Both primary options are affected, and both produce a stylesheet that no longer parses.

### What minimal example or steps are needed to reproduce the bug?

```scss
a {
	color // keep me
	: red;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-scss",
  "plugins": "@stylistic/stylelint-plugin",
  "rules": {
    "@stylistic/declaration-colon-space-before": "never"
  }
}
```

The `always` option reproduces it just as well.

### What did you expect to happen?

An inline comment before the colon makes either option impossible to satisfy: whatever is put on that line after the comment becomes part of the comment. The fix should leave the code alone and let the warning stand, rather than rewrite the source into something that no longer parses.

### What actually happened?

With `never`:

```scss
a {
	color // keep me: red;
}
```

With `always`:

```scss
a {
	color // keep me : red;
}
```

In both the colon and the value are now inside the comment, and parsing the output back fails with `<css input>:2:2: Unknown word color`.

The cause is at [`lib/rules/declaration-colon-space-before/index.js:52`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/rules/declaration-colon-space-before/index.js#L52) and [`:58`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/rules/declaration-colon-space-before/index.js#L58):

```js
decl.raws.between = between.slice(0, colonIndex).replace(/\s*$/u, ` `) + between.slice(colonIndex)
decl.raws.between = between.slice(0, colonIndex).replace(/\s*$/u, ``) + between.slice(colonIndex)
```

`\s*$` treats the newline that closes the `//` comment as ordinary trailing whitespace. The point replacement is otherwise right: a block comment is preserved exactly as it should be, `color /* keep me */ : red` becoming `color /* keep me */: red`. Only the inline form is affected.

`declaration-colon-space-after` was checked as well and is **not** affected: the whitespace it touches sits between the colon and any comment that follows, so removing it cannot swallow anything. `indentation` and `no-eol-whitespace` are not affected either — neither of them deletes a newline.

### Do you have a proposal to fix the bug?

The same shape as the fix for #63: cut the trailing whitespace run off the part before the colon, and when what remains ends with an inline comment, leave that part as it was and let the warning stand.

Once a second rule needs this reasoning — #87 and this one do — the check for “the content ends with an inline comment” is worth extracting into a shared helper in `lib/utils/` instead of being spelled out in every rule.
